Day-2/problem-1.go

```diff
@@ -4,51 +4,58 @@ import (
 	"fmt"
 	"strings"
 	"sync"
-	"time"
 )
 
+type Counter struct {
+	mutex      sync.Mutex
+	wordArray  []string
+	letterFreq map[string]int
+}
+
 // Init initialises the frequency map for 26 characters
-func Init(letterFreq map[string]int) {
+func (counter *Counter) Init() {
+	counter.letterFreq = make(map[string]int)
 	for char := 'a'; char <= 'z'; char++ {
-		letterFreq[string(char)] = 0
+		counter.letterFreq[string(char)] = 0
 	}
 }
 
 // preProcess transforms the string to lower case
-func preProcess(wordArray []string) {
-	for _, word := range wordArray {
-		word = strings.ToLower(word)
+func (counter *Counter) preProcess() {
+	for id, word := range counter.wordArray {
+		counter.wordArray[id] = strings.ToLower(word)
 	}
 }
 
 // calcLetterFrequency calculates the frequency of each letter
-func calcLetterFrequency(wordArray []string, letterFreq map[string]int, mutex *sync.Mutex, wg *sync.WaitGroup) {
-	for _, word := range wordArray {
+func (counter *Counter) calcLetterFrequency(wg *sync.WaitGroup) {
+	for _, word := range counter.wordArray {
 		wg.Add(1)
```

Comment:
Could have used for loop in the main function and sent every string over a channel  and therefore using wg.Add(1) in main function.

Reply:
I think it's all good since I am not using go routine in main thread for calling calculateFrequency method